File: Leetcode/BinarySearch/C++/searchMatrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n_rows = matrix.size(), n_cols = matrix[0].size();
        if (n_rows < 0 || n_rows > 100 || n_cols < 0 || n_cols > 100) {
            return false;
        }

        int i = 0, j = n_cols - 1;
        while (i < n_rows && j >= 0) {
            if (target == matrix[i][j]) {
                return true;
            }
            else if (target < matrix[i][j]) {
                j--;
            }
            else {
                i++;
            }
        }

        return false;
    }
};

File: Leetcode/Math/Python/search_matrix.py

def search_matrix(matrix):

    res = 0
    n_rows = len(matrix)
    n_cols = len(matrix[0])

    zeros_mat = [0 for i in range(n_rows) for i in range(n_cols)]

    for i in range(n_rows):
        for j in range(n_cols):

            if i == 0 or j == 0:
                zeros_mat[i][j] = int(matrix[i][j])

            else:
                if matrix[i][j] =! '0':
                    
            



